### ISSUE TITLE
Getting completion for module that is not in "public"

### ISSUE THREAD

**Issue by aspeddro** [rescript-lang/rescript-vscode#831]
Using latest-master. Related #824

![image](https://github.com/rescript-lang/rescript-vscode/assets/16160544/6b50d8c9-90c5-443b-9a49-ee434e36f0f5)

Repo https://github.com/aspeddro/rescript-docgen-test

**Comment by zth** [comment]
Yes, I implemented the public check wrongly, at the top level instead of in each subdir. Need to re-implement that, but currently swamped. Let me know if you're interested in taking a stab at it, and I can help guide you.
